### PR TITLE
RAC-944: Add Job Execution Table

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

- Add a job execution table with pagination in job execution list page
- Make Pagination DSM component sticky

**Definition Of Done (for Core Developer only)**

- [x] Tests

### CODE REVIEW

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 1]
```diff
@@ -73,17 +78,28 @@ public function count(SearchJobExecutionQuery $query): int
...
+        $platform = $this->connection->getDatabasePlatform();
+
+        return array_map(function (array $rawJobExecution) use ($platform): JobExecutionRow {
+            $errors = json_decode($rawJobExecution['errors'], true); // TODO revalidate that currently the errors are here
+            $errorCount = 0;
+            foreach ($errors as $error) {
+                $errorCount += count(unserialize($error));
+            }
+
+            $startTime = Type::getType(Types::DATETIME_MUTABLE)->convertToPHPValue($rawJobExecution['start_time'], $platform);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:90]:
```suggestion
            $startTime = Type::getType(Types::DATETIME_IMMUTABLE)->convertToPHPValue($rawJobExecution['start_time'], $platform);
```

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import {JobExecutionRow} from '../../models/JobExecutionTable';
+import {JobExecutionStatus} from '../JobExecutionStatus';
+
+const JobExecutionTable = ({jobExecutionRows}: {jobExecutionRows: JobExecutionRow[]}) => {
+  const translate = useTranslate();
+  const dateFormatter = useDateFormatter();
+
+  return (
+    <Table>
+      <Table.Header sticky={25 < jobExecutionRows.length ? 44 : 0}>
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:13]: NO

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+import {JobExecutionTable} from '../components/JobExecutionList/JobExecutionTable';
 
 const JobExecutionList = () => {
   const translate = useTranslate();
-  const searchJobExecutionTableResult = useSearchJobExecutionTableResult();
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const jobExecutionTable = useJobExecutionTable(currentPage);
   const activityHref = useRoute('pim_dashboard_index');
-  const resultMatches = searchJobExecutionTableResult === null ? 0 : searchJobExecutionTableResult.matches_count;
+  const jobExecutionMatches = jobExecutionTable === null ? 0 : jobExecutionTable.matches_count;
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx:12]: Maybe 

```suggestion
  const matchesCount = jobExecutionTable === null ? 0 : jobExecutionTable.matches_count;
```

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.test.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  },
+];
+
+test('it renders a Job execution Table', () => {
+  renderWithProviders(<JobExecutionTable jobExecutionRows={rows} />);
+
+  expect(screen.getByText('Nice name')).toBeInTheDocument();
+  expect(screen.getByText('pim_import_export.widget.last_operations.job_type.export')).toBeInTheDocument();
+  expect(screen.getByText('admin')).toBeInTheDocument();
+  expect(screen.getByText('pim_import_export.job_status.STARTED 1/2')).toBeInTheDocument();
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.test.tsx:30]:
```suggestion
  expect(screen.getByText('pim_import_export.job_status.STARTED 1/2')).toBeInTheDocument();
  expect(screen.getByText('4')).toBeInTheDocument();
```

File: front-packages/shared/src/hooks/useDateFormatter.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import {useCallback} from 'react';
+import {useUserContext} from "./useUserContext";
+
+export const useDateFormatter = () => {
```
Comment (StevenVAIDIE) [inline front-packages/shared/src/hooks/useDateFormatter.ts:4]: Missing tests

File: src/Akeneo/Platform/Job/back/Application/SearchJobExecution/JobExecutionRow.php [hunk 6]
```diff
@@ -11,32 +11,56 @@
...
     }
 
     public function normalize(): array
     {
         return [
+            'job_execution_id' => $this->jobExecutionId,
             'job_name' => $this->jobName,
             'type' => $this->type,
-            'start_at' => $this->startAt,
+            'started_at' => $this->startedAt ? $this->startedAt->format(DATE_ATOM) : null,
```
Comment (pierrejolly) [inline src/Akeneo/Platform/Job/back/Application/SearchJobExecution/JobExecutionRow.php:55]: What about timezone?

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx (hunk 2 above)
Comment (ValentinMumble) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:13]: YES

**Review by ValentinMumble — APPROVED** [review]

**Review by ghoussard — APPROVED** [review]

**Pull request merged**